### PR TITLE
Replace xtend with builtin Object.assign

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/apostrophecms/sanitize-html/issues/257

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -101,9 +100,9 @@ function sanitizeHtml(html, options, _recursing) {
     options = sanitizeHtml.defaults;
     options.parser = htmlParserDefaults;
   } else {
-    options = extend(sanitizeHtml.defaults, options);
+    options = Object.assign({}, sanitizeHtml.defaults, options);
```
Comment (abea) [inline src/index.js:103]: I think we can let `sanitizeHtml.defaults` be the target argument since it's a set of defaults.

File: src/index.js [hunk 2]
```diff
@@ -101,9 +100,9 @@ function sanitizeHtml(html, options, _recursing) {
...
     if (options.parser) {
-      options.parser = extend(htmlParserDefaults, options.parser);
+      options.parser = Object.assign({}, htmlParserDefaults, options.parser);
```
Comment (abea) [inline src/index.js:105]: Same here, making `htmlParserDefaults` the target argument.

**Comment by TrySound** [comment]
Mutating defaults does not look right. I used Object.assign in immutable way similar to default xtend implementation.
https://github.com/Raynos/xtend/blob/master/package.json#L15

**Comment by TrySound** [comment]
This is especially wrong because you expose defaults to the user.

**Comment by TrySound** [comment]
Object.assign mutates target object by the way.

**Comment by TrySound** [comment]
@abea Rebased and fixed base branch.

**Pull request merged**